channels/rdpdr/client/rdpdr_main.c

```diff
@@ -1652,6 +1652,13 @@ static UINT rdpdr_virtual_channel_event_disconnected(rdpdrPlugin* rdpdr)
 {
 	UINT error;
 
+	/* Remove all dependent devices before tearing down resources */
+	if (rdpdr->devman)
```

Comment:
Are you sure this does not add a race condition? Would need to check that my self in depth, but I assume you just want to clear the device list first?